docs/configuration.rst

```diff
@@ -986,6 +986,134 @@ remain available as described at the start of the :ref:`Configuration` section.
     Name of the :term:`Provider` used for login. This represents the identifier that is set to define how to
     differentiate between a local sign-in procedure and a dispatched one some known :ref:`authn_providers`.
 
+Network Mode Settings
+~~~~~~~~~~~~~~~~~~~~~
+
+The following configuration parameters are related to Magpie's "Network Mode" which allows networked instances of Magpie
+to authenticate users for each other. All variables defined in this section are only used if
+:envvar:`MAGPIE_NETWORK_ENABLED` is enabled.
+
+.. envvar:: MAGPIE_NETWORK_ENABLED
+
+    [:class:`bool`]
+    (Default: ``False``)
+
+    .. versionadded:: 3.38
+
+    Enable "Network Mode" which enables all functionality to authenticate users using other Magpie instances as
+    external authentication providers.
+
+.. envvar:: MAGPIE_NETWORK_INSTANCE_NAME
+
+    [:class:`str`]
+
+    .. versionadded:: 3.38
+
+    The name of this Magpie instance in the network. This variable is used to determine if an authentication token was
+    issued by this instance of Magpie, or another instance in the network.
+
+    This variable is required if :envvar:`MAGPIE_NETWORK_ENABLED` is ``True``.
+
+.. envvar:: MAGPIE_NETWORK_DEFAULT_TOKEN_EXPIRY
+
+    [:class:`int`]
+    (Default: ``86400``)
+
+    .. versionadded:: 3.38
+
+    The default expiry time (in seconds) for an access token issued for the purpose of network authentication.
+
+.. envvar:: MAGPIE_NETWORK_INTERNAL_TOKEN_EXPIRY
+
+    [:class:`int`]
+    (Default: ``30``)
+
+    .. versionadded:: 3.38
+
+    The default expiry time (in seconds) for an JSON Web Token issued for the purpose of communication between nodes in
+    the network.
+
+.. envvar:: MAGPIE_NETWORK_TOKEN_NAME
+
+    [|constant|_]
+    (Value: ``"magpie_token"``)
+
+    .. versionadded:: 3.38
+
+    The name of the request parameter key whose value is the authentication token issued for the purpose of network
+    authentication.
+
+.. envvar:: MAGPIE_NETWORK_PROVIDER
+
+    [|constant|_]
+    (Value: ``"magpie_network"``)
+
+    .. versionadded:: 3.38
+
+    The name of the external provider that authenticates users using other Magpie instances as external authentication
+    providers.
+
+.. envvar:: MAGPIE_NETWORK_NAME_PREFIX
+
+    [|constant|_]
+    (Value: ``"anonymous_network_"``)
+
+    .. versionadded:: 3.38
+
+    A prefix added to the anonymous network user and network group names. These names are constructed by prepending the
```

Comment:
Reuse term `Network User`.

Should there be a `Network Group` term as well? It seems only related to `MAGPIE_NETWORK_GROUP_NAME`. Is that correct? What is its implication/usage, and could it be needed to have different onces per node?

Reply:
> Should there be a Network Group term as well?

There could be. We're currently not referring to it anywhere else in the documentation but we could put it in for the future.

> What is its implication/usage

It's a group that gets created for each network node. Each user who logs in from that network node is part of the group. So if you want to have different access permissions for people from node A vs node B you'd set those permissions on the appropriate network group.

>  could it be needed to have different onces per node?

No, only one per node

Reply:
Actually thinking about it again, there's no real need for a definition since it's not functionally different from a regular group and I think it would be confusing to give it a special definition.